src/engine.rs

```diff
@@ -698,40 +514,24 @@ impl Reedline {
             ]
```

Comment:
(Also pinging @mzanrosso) should the addition to the undo stack be completely encapsulated by the `Editor` or do we envision cases where it would be better that the commands on the editor don't do it automatically to not get confused when plugins/completions start to edit?

Reply:
I quickly saw the changes and I think that currently nothing changes if we keep it on linebuffer or editor;  I should spend more time analyzing the code well to respond with a bit of foresight